packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -159,108 +161,170 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
+      final File templateFile = globals.fs.file(
+          globals.fs.path.join(templateImageDirName, templateFileName));
+
+      return projectFile.existsSync()
+          && templateFile.existsSync()
+          && md5.convert(projectFile.readAsBytesSync()) ==
+              md5.convert(templateFile.readAsBytesSync());
+    });
+  }
 
+  // A helper function to return a list of image files in an image asset with
+  // wrong sizes (as specified in its Contents.json file).
+  List<String> _imageFilesWithWrongSize({
+    required Map<_ImageAssetFileKey, String> imageInfoMap,
+    required String imageDirName,
+  }) {
+    return imageInfoMap.entries.where((MapEntry<_ImageAssetFileKey, String> entry) {
+      final String fileName = entry.value;
+      final File imageFile = globals.fs.file(globals.fs.path.join(imageDirName, fileName));
+      if (!imageFile.existsSync()) {
+        return false;
+      }
       // validate image size is correct.
       // PNG file's width is at byte [16, 20), and height is at byte [20, 24), in big endian format.
       // Based on https://en.wikipedia.org/wiki/Portable_Network_Graphics#File_format
-      final ByteData projectIconData = projectIconBytes.buffer.asByteData();
-      if (projectIconData.lengthInBytes < 24) {
-        continue;
-      }
-      final int width = projectIconData.getInt32(16);
-      final int height = projectIconData.getInt32(20);
-      if (width != entry.key.pixelSize || height != entry.key.pixelSize) {
-        filesWithWrongSize.add(entry.value);
+      final ByteData imageData = imageFile.readAsBytesSync().buffer.asByteData();
+      if (imageData.lengthInBytes < 24) {
+        return false;
       }
-    }
+      final int width = imageData.getInt32(16);
+      final int height = imageData.getInt32(20);
+      // The size must not be null.
+      final int expectedSize = entry.key.pixelSize!;
+      return width != expectedSize || height != expectedSize;
+    })
+    .map((MapEntry<_ImageAssetFileKey, String> entry) => entry.value)
+    .toList();
+  }
 
-    if (filesWithTemplateIcon.isNotEmpty) {
+  Future<void> _validateIconAssetsAfterArchive(StringBuffer messageBuffer) async {
+    final BuildableIOSApp app = await buildableIOSApp;
+
+    final Map<_ImageAssetFileKey, String> templateInfoMap = _parseImageAssetContentsJson(
+      app.templateAppIconDirNameForContentsJson,
+      requiresSize: true);
+    final Map<_ImageAssetFileKey, String> projectInfoMap = _parseImageAssetContentsJson(
+      app.projectAppIconDirName,
+      requiresSize: true);
+
+    final bool usesTemplate = _isAssetStillUsingTemplateFiles(
+      templateImageInfoMap: templateInfoMap,
+      projectImageInfoMap: projectInfoMap,
+      templateImageDirName: await app.templateAppIconDirNameForImages,
+      projectImageDirName: app.projectAppIconDirName);
+    if (usesTemplate) {
       messageBuffer.writeln('\nWarning: App icon is set to the default placeholder icon. Replace with unique icons.');
```

Comment:
I know this is from a previous PR, but I'm wondering if it will be obvious to the user where to update this, or if we should give them more instruction?

Reply:
yep, the same instruction is shared by all validations (they are on the same box): 
![Screen Shot 2022-11-29 at 10 40 57 PM](https://user-images.githubusercontent.com/41930132/205187632-f15e077e-8a8a-4f0a-9ff4-d255f28a7c4f.png)

Reply:
Ahh, I missed the docs link, that's looks good